main/MainCamera.gd

```diff
@@ -0,0 +1,50 @@
+extends Camera2D  
+
+var last_pos
+var target_zoom = 1.0
+var zoom_rate = 5.0
+const MIN_ZOOM = 0.2
+const MAX_ZOOM = 2.0
+const ZOOM_INCREMENT = 0.1
+
+func _ready():
+	last_pos = position
+
+func _unhandled_input(event: InputEvent) -> void:
+	if is_current():  
+		if event is InputEventMouseMotion:
+			if event.button_mask == MOUSE_BUTTON_MASK_LEFT:
+				position -= event.relative
```

Comment:
The applied position offset should scale with the zoom level as it feels like you drag the camera over a sticky surface.

Reply:
Had a second look, apparently it already does it, but it still is not enough.